### ISSUE TITLE
%stacktrace / %stacktracedetail of log4net is not working

### ISSUE THREAD

**Issue by HeikoWitthoeft** [csnemes/tracer#21]
If i enable the property %stacktrace / %stacktracedetail for my traceappender I get an log4net.ERROR:

log4net:ERROR loggingEvent.LocationInformation.StackFrames was null or empty.

And all messages generated by tracer (Method Enter and Exit) do not contain the information of the stacktrace. My own calls to log4netLogger.Trace do contain the stacktrace like I want it.
Is there any way to enable Tracer to add this? Or is there just something wrong with my setup here?

**Comment by csnemes** [comment]
Hi,
This behaviour (i.e. stack info is missing) is by design. The reason that logging via this rewriter can be significantly faster than the original log4net is that it don't create stack frames when logging. The class and method information is 'calculated' rewrite time and woven into the code. Creating stack frames is an expensive procedure which can be saved this way. The downside of course is that you don't have StackFrames (nor FileName or LineNumber) in the LocationInfo. Could you please describe your use case to see if I can come up with some alternative solution.

**Comment by HeikoWitthoeft** [comment]
Hey,

thanks a lot for the fast reply.
I am just exploring solutions to get tracing enabled for all of our projects without adding any Attribute to it so I tried using tracer and was just wondering about the missing possibility of a stacktrace.

We want to be able to get a detailed trace logging of our application, just for the case of debugging or peoblems occuring only on one installation of it e.g.

Would this stacktrace only be generated if the trace logging is enabled / used or even if tracing is disabled? Then I totaly agree that this might be not a good Idea. Might it be a good idea to have this configurable? (Then I could adjust the build if i need detail information)

**Comment by csnemes** [comment]
Stacktrace is only generated when the logging event is created in log4net, so if the trace level is disabled it won't be created at all. We use this tool in an always on fashion thus the speed is very important for us. The way how tracer uses log4net is in the adapter (not in the rewriter itself) so it would be possible (in fact this was one of the main goals) to write/modify new/existing adapters easily. I'll take a look on how can I put a switch to turn on/off stack frame creation or even create a separate log4net adapter which provides stack frames.

**Comment by HeikoWitthoeft** [comment]
Hey, thanks again for your reply. Wih that hint I can maybe (or better hopefully?) do it by my own. I will check the implementation of the log4net adapter tomorrow and will come back to you if there is something unclear for me. Maybe I can share my solution to you afterwards.

**Comment by HeikoWitthoeft** [comment]
Hey csnemes,
I just tried yesterday our framework with Tracer inside and enabled some classes. Over the Thread ID and the Enter/Leave logging we have indirectly the information of where the method call came from. And if a exception is caught the full stacktrace is put into the Trace information (I did this for testing once).

So right now we decided like you in the implementation - performance over information overkill. Thanks again for your good implementation and also for the support here!

Best Regards
